front/pages/api/poke/workspaces/[wId]/invitations.ts

```diff
@@ -89,24 +143,42 @@ async function handler(
         owner.sId
       );
 
-      const invitationRes = await handleMembershipInvitations(
-        workspaceAdminAuth,
-        {
+      const previousMember = await getExistingMember(
+        owner,
+        bodyValidation.right.email
+      );
+
+      if (previousMember) {
+        const membershipUpdateRes = await handleMembershipUpdate(
           owner,
-          user,
-          subscription,
-          invitationRequests: [bodyValidation.right],
+          previousMember,
+          bodyValidation.right.role
+        );
+        if (membershipUpdateRes.isErr()) {
+          return apiError(req, res, membershipUpdateRes.error);
         }
-      );
 
-      if (invitationRes.isErr()) {
-        return apiError(req, res, invitationRes.error);
-      }
+        res.status(200).json(membershipUpdateRes.value);
+        return;
+      } else {
+        const invitationRes = await handleMembershipInvitations(
```

Comment:
Why not fixing `handleMembershipInvitations` directly? This function should support re-inviting a revoked member. Even though today the UI does not leverage this endpoint for this IMO either it should error or update the role.